packages/login-ui/redux/utils.js

```diff
@@ -0,0 +1,53 @@
+import { getNetworkById, getNetworksSelectorGroup } from '@chronobank/login/network/settings'
+import web3Factory from '@chronobank/core/web3'
+import { compact } from 'lodash'
+
+const check = async (network) => {
+  const w3 = web3Factory(network)
+  try{
+    return await w3.eth.net.isListening()
+  } catch(e){
+   return false
+  }
+}
+
+ export async function findProvider (providersList){
+  let num = 0
+  while (providersList[num]) {
+    const netData = {
+      selectedNetworkId: providersList[num].network.id,
+      selectedProviderId: providersList[num].provider.id,
+    }
+    const network = getNetworkById(netData.selectedNetworkId, netData.selectedProviderId)
+
+    if (await check(network)) return netData
```

Comment:
It's just my opinion , but using await in if condition looks very strange. Let's use intermediate parameter upper:
```
const isNetworkWorks = await check(network)
if (isNetworkWorks) {
  return netData
}
```